fusor-ember-cli/app/controllers/review/installation.js

```diff
@@ -242,33 +242,18 @@ export default Ember.Controller.extend(NeedsDeploymentMixin, OpenshiftMixin, {
     return this.get('storageSize') + ' GB';
   }),
 
-  deploymentButtonAction: Ember.computed('hasSubscriptionsToAttach', function() {
-    if (this.get('showWarningMessage')) {
-        return "showContinueDeployModal";
-    } else if (this.get('hasSubscriptionsToAttach')) {
-        return "attachSubscriptions";
-        return "installDeployment";
-    } else if (this.get('showWarningMessage')) {
-        return "showContinueDeployModal";
-    }
-  }),
-
   closeContinueDeployModal() {
-    this.set('continueDeploymentModalOpen', false);
-    this.set('continueDeploymentModalClosed', true);
-    this.set('modalOpen', false);
+    this.set('openModal', false);
   },
 
   actions: {
     showContinueDeployModal() {
       this.set('deploymentInModal', this.get('model'));
-      this.set('continueDeploymentModalOpen', true);
-      this.set('continueDeploymentModalClosed', false);
-      this.set('modalOpen', true);
+      this.set('openModal', true);
     },
 
     onDeployButton() {
-      if (this.get('showWarningMessage')) {
+      if (this.get('showValidationWarnings') || this.get('showValidationErrors')) {
```

Comment:
Showing the deployment warning model because `showValidationErrors` is true is a scenario that should not happen, because the deployment button should not be enabled if there are validation errors on the deployment. It's semantically incorrect.

We're actually already trying to do this with `buttonDeployDisabled`, but there's a bug in it since it's missing a dependency on `showValidationErrors`. [This patch](https://gist.githubusercontent.com/eriknelson/0b32b984c0c28fac12c4873d1fc8c3f4/raw/2fa87313be905a9184082606d4d3035a1ea37ca1/gistfile1.txt) should fix the bug and then we can remove the redundant `showValidationErrors` or condition.